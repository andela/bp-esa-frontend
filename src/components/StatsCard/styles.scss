.stats-wrapper {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 100%;
  padding: 0 49px 0 49px;

  .stat-container {
    display: flex;
    flex-direction: row;
    box-sizing: border-box;
    height: 96px;
    width: 210.5px;
    border-radius: 5px;
    background-color: #FFF;
    border: 1px solid #CCD;

    .stat-icon {
      height: 96px;
      width: 106.67px;
      border-radius: 5px 0 0 5px;
      background-color: rgba(204,204,221,0.2);
      text-align: center;

      img {
        margin-top: 20px;
      }

      p {
        height: 18px;
        width: 107px;
        margin-top: 6px;
        color: #3359DF;
        font-family: Arial;
        font-size: 16px;
        line-height: 18px;
        text-align: center;
      }
    }

    .stat-content {
      width: 103.83px;
      text-align: center;

      path.color0 {
        fill: #43A047;
      }
      
      path.color1 {
        fill: #D32F2F;
      }
      
      text {
        font-family: 'DIN pro';
        font-size: 18px;
        font-weight: 400;
        line-height: 16rem;
        fill: #3359DF;
      }
    }
  }
  .loader-container {
    margin: auto;
  }
}
