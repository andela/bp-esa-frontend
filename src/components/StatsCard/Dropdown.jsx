import React, { PureComponent } from 'react';
import PropTypes from 'proptypes';
import ReactDOM from 'react-dom';
import './styles.scss';

class Dropdown extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      collapse: true,
      categoryTitle: 'today',
    };
  }

  showSection = () => {
    const { collapse } = this.state;
    const { collapseValue } = this.hideSection(collapse);
    this.setState({
      collapse: collapseValue,
    });
    document.addEventListener('click', this.closeStatDropdown);
  };

  hideSection = (collapse) => {
    let collapseValue;
    if (!collapse) {
      collapseValue = true;
    } else {
      collapseValue = false;
    }
    return { collapseValue };
  };

  closeStatDropdown = (event) => {
    const statsDropDown = ReactDOM.findDOMNode(this);
    if (event.target !== statsDropDown) {
      this.setState({
        collapse: true,
      });
      document.removeEventListener('click', this.closeStatDropdown);
    }
  };

  selectPeriod = (event) => {
    const { fetchStat } = this.props;
    const { innerHTML, id } = event.target;
    this.setState({
      collapse: true,
      categoryTitle: innerHTML,
    });
    fetchStat(id);
  }

  render() {
    const { collapse, categoryTitle } = this.state;
    return (
      <div className="stat-dropdown-container">
        Automations stats for
        <span className="stat-dropdown-caret" onClick={this.showSection}>
          {` ${categoryTitle}`}<img src="/drop-down.png" alt="down" />
        </span>
        {!collapse ? (
          <React.Fragment>
            <div className="stat-dropdown-content">
              <div className="stat-dropdown-item" id="days" onClick={this.selectPeriod}>
                today
              </div>
              <div className="stat-dropdown-item" id="weeks" onClick={this.selectPeriod}>
                this week
              </div>
              <div className="stat-dropdown-item" id="months" onClick={this.selectPeriod}>
                this month
              </div>
              <div className="stat-dropdown-item" id="years" onClick={this.selectPeriod}>
                this year
              </div>
            </div>
          </React.Fragment>
        ) : null}
      </div>
    );
  }
}

Dropdown.propTypes = {
  fetchStat: PropTypes.func.isRequired,
};

export default Dropdown;
