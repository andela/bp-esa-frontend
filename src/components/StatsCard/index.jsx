import React, { PureComponent } from 'react';
import PropTypes from 'proptypes';
import createDonut from '../../utils/automationStat';
import capitalizeFirstLetter from '../../utils/capitalize';
import './styles.scss';

class StatsCard extends PureComponent {
  componentDidMount() {
    const { type, stat } = this.props;
    createDonut(stat, `${type}`);
  }

  render() {
    const { type } = this.props;
    return (
      <div className="stat-container">
        <div className="stat-icon">
          <img src={`/${capitalizeFirstLetter(type)}.png`} alt="icon" />
          <p>{capitalizeFirstLetter(type)}</p>
        </div>
        <div id={`${type}`} className="stat-content" />
      </div>
    );
  }
}

StatsCard.propTypes = {
  stat: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
};

export default StatsCard;
