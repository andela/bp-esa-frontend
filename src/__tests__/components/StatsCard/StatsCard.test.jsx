import React from 'react';
import { shallow } from 'enzyme';
import StatsCard from '../../../components/StatsCard';

const props = {
  stat: {
    success: 1,
    total: 191,
  },
  type: 'Freckle',
};

const getComponent = () => shallow(<StatsCard {...props} />);

describe('StatsCard', () => {
  it('should render as expected', () => {
    expect(getComponent()).toMatchSnapshot();
  });
});
