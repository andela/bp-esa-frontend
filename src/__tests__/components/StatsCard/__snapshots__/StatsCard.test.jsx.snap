// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsCard should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StatsCard
    stat={
      Object {
        "success": 1,
        "total": 191,
      }
    }
    type="Freckle"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="stat-icon"
        >
          <img
            alt="icon"
            src="/Freckle.png"
          />
          <p>
            Freckle
          </p>
        </div>,
        <div
          className="stat-content"
          id="Freckle"
        />,
      ],
      "className": "stat-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="icon"
              src="/Freckle.png"
            />,
            <p>
              Freckle
            </p>,
          ],
          "className": "stat-icon",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "icon",
              "src": "/Freckle.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Freckle",
            },
            "ref": null,
            "rendered": "Freckle",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "stat-content",
          "id": "Freckle",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="stat-icon"
          >
            <img
              alt="icon"
              src="/Freckle.png"
            />
            <p>
              Freckle
            </p>
          </div>,
          <div
            className="stat-content"
            id="Freckle"
          />,
        ],
        "className": "stat-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="icon"
                src="/Freckle.png"
              />,
              <p>
                Freckle
              </p>,
            ],
            "className": "stat-icon",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "icon",
                "src": "/Freckle.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Freckle",
              },
              "ref": null,
              "rendered": "Freckle",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stat-content",
            "id": "Freckle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
